src/linker/Linker.Dataflow/XmlFlowAnnotationSource.cs

```diff
@@ -28,76 +27,121 @@ public XmlFlowAnnotationSource (LinkContext context)
+					continue;
+				}
+				string[] xmlArguments = arguments.ToArray ();
+				bool recognizedArgument = true;
 
-		public DynamicallyAccessedMemberTypes GetPropertyAnnotation (PropertyDefinition property)
-		{
-			return _properties.TryGetValue (property, out var ann) ? ann : DynamicallyAccessedMemberTypes.None;
-		}
+				CustomAttribute attribute = new CustomAttribute (constructor);
+				for (int i = 0; i < xmlArguments.Length; i++) {
+					object argumentValue = null;
 
-		public DynamicallyAccessedMemberTypes GetReturnParameterAnnotation (MethodDefinition method)
-		{
-			return _methods.TryGetValue (method, out var ann) ? ann.ReturnAnnotation : DynamicallyAccessedMemberTypes.None;
+					if (constructor.Parameters[i].ParameterType.Resolve ().IsEnum) {
+						foreach (var field in constructor.Parameters[i].ParameterType.Resolve ().Fields) {
+							if (field.IsStatic && field.Name == xmlArguments[i]) {
+								argumentValue = Convert.ToInt32 (field.Constant);
+								break;
+							}
+						}
+						if (argumentValue == null) {
+							_context.LogMessage (MessageContainer.CreateWarningMessage ($"Could not parse argument '{xmlArguments[i]}' specified in '{_xmlDocumentLocation}' as a {constructor.Parameters[i].ParameterType.FullName}", 2021));
+							recognizedArgument = false;
+						}
+					} else {
+						switch (constructor.Parameters[i].ParameterType.MetadataType) {
+						case MetadataType.String:
+							argumentValue = xmlArguments[i];
+							break;
+						case MetadataType.Int32:
+							argumentValue = int.Parse (xmlArguments[i]);
+							break;
+						default:
+							_context.LogMessage (MessageContainer.CreateWarningMessage ($"Argument '{xmlArguments[i]}' specified in '{_xmlDocumentLocation}' could not be transformed to a currently supported metadatatype in the xml", 2020));
+							recognizedArgument = false;
+							break;
+						}
+					}
+					attribute.ConstructorArguments.Add (new CustomAttributeArgument (constructor.Parameters[i].ParameterType, argumentValue));
+				}
+				if (recognizedArgument)
+					attributes.Add (attribute);
+			}
+			return attributes;
 		}
 
-		public DynamicallyAccessedMemberTypes GetThisParameterAnnotation (MethodDefinition method)
+		public TypeDefinition GetTypeFromName (string fullName)
 		{
-			if (_methods.TryGetValue (method, out var ann) && ann.ParameterAnnotations != null) {
-				foreach (var (ParamName, Annotation) in ann.ParameterAnnotations)
-					if (ParamName == "this")
-						return Annotation;
-			}
+			TypeDefinition attributeType;
+			foreach (AssemblyDefinition asm in _context.GetAssemblies ()) {
+				attributeType = asm.FindType (fullName);
+				if (attributeType != null)
+					return attributeType;
 
-			return DynamicallyAccessedMemberTypes.None;
+				foreach (AssemblyDefinition refasm in _context.ResolveReferences (asm)) {
```

Comment:
I would expect all referenced assemblies to be resolved by the time we get here... thus we should not need this loop.